src/printer/smt2/smt2_printer.cpp

```diff
@@ -509,7 +515,8 @@ void Smt2Printer::toStream(std::ostream& out,
       // the logic is mixed int/real. The former occurs more frequently.
       bool is_int = force_nt.isInteger();
       // If constant rational, print as special case
-      if (type_asc_arg.getKind() == kind::CONST_RATIONAL)
+      Kind ka = type_asc_arg.getKind();
+      if (ka == kind::CONST_RATIONAL || ka == kind::CONST_INTEGER)
```

Comment:
This essentially does subtyping here? Should we set `is_int` to `ka == kind::CONST_INTEGER` instead?

Reply:
This code will be deleted soon after this PR.